indico/modules/rb/client/js/modules/calendar/CalendarListView.jsx

```diff
@@ -101,23 +128,86 @@ class CalendarListView extends React.Component {
           {moment(day, 'YYYY-MM-DD').format('dddd, LL')}
         </Header>
         <Card.Group itemsPerRow={4} stackable styleName="day-cards-container">
-          {bookings.map(this.renderBooking)}
+          {bookings.map(booking => this.renderBooking(booking, day, linkData))}
         </Card.Group>
         <Divider hidden />
       </div>
     );
   };
 
-  renderBooking = booking => {
+  renderLink = (booking, day, linkData, linkBookingOccurrence) => {
+    if (linkData) {
+      const {linkingConfirmOpen} = this.state;
+      const {reservation} = booking;
+      const boundaries = [moment(linkData.startDt), moment(linkData.endDt)];
+      const datesMatch =
+        moment(booking.startDt).isBetween(...boundaries, undefined, '[]') &&
+        moment(booking.endDt).isBetween(...boundaries, undefined, '[]');
+      if (booking.linkId || !datesMatch) {
+        return;
+      }
+      const linkBtn = (
+        <Button
+          icon={<Icon name="linkify" />}
+          primary
+          size="small"
+          circular
+          onClick={e => {
+            e.stopPropagation();
+            this.setState({linkingConfirmOpen: true});
+          }}
+        />
+      );
+      return (
+        <>
+          <Confirm
+            size="mini"
+            header={Translate.string('Link event')}
+            open={linkingConfirmOpen}
```

Comment:
This renders many Confirm prompts, instead of only the confirm prompt for the occurrence you're actually trying to link. So when confirming, you can easily linking the wrong booking (I think if you have two on the same day, you always end up linking the last one, but that may be pure coincidence, not sure how strict React is when it comes to render order).

Reply:
Fixed in my latest commit